src/Orc.CsvHelper/Orc.CsvHelper.Shared/Extensions/TypeExtensions.cs

```diff
@@ -0,0 +1,25 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="TypeExtensions.cs" company="WildGums">
+//   Copyright (c) 2008 - 2017 WildGums. All rights reserved.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+
+namespace Orc.Csv
+{
+    using System;
+    using Catel;
+
+    internal static class TypeExtensions
+    {
+        #region Methods
+        public static T CreateInstanceOfType<T>(this Type type)
```

Comment:
but we will store reference for it in static class, otherwise we need to resolve it any time (or pass it as parameter)...